crates/core/src/masp.rs

```diff
@@ -238,6 +238,39 @@ pub fn encode_asset_type(
     .encode()
 }
 
+/// Encode the assets that are used for masp rewards
+pub fn encode_reward_asset_types(
+    native_token: &Address,
+) -> Result<[AssetType; 4], std::io::Error> {
```

Comment:
Rename to `encode_native_token_reward_asset_types`? Or make it explicit in the docstring that this function is only supposed to be called with the native token address. This will improve the docs published to `docs.rs`.